### PR TITLE
Feature/csv persistence

### PR DESCRIPTION
**Pull request by romaninsh**
Still work in progress.

### CODE REVIEW

**Comment by romaninsh** [comment]
Documentation is missing.

File: src/Persistence_CSV.php [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    public function insert(Model $m, $data)
+    {
+        if (!$this->mode) {
+            $this->mode = 'w';
+        } elseif ($this->mode == 'r') {
+            throw new Exception(['Currently reading records, so writing is not possible.']);
+        }
+
+        if (!$this->handle) {
+            $this->saveHeader();
```
Comment (DarkSide666) [inline src/Persistence_CSV.php]: no such method exist

File: src/Persistence_CSV.php [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        }
+
+        $id = $this->generateNewID($m, $table);
```
Comment (DarkSide666) [inline src/Persistence_CSV.php]: $table doesn't exist

**Comment by DarkSide666** [comment]
Documentation is there, but there is a lot missing in loading and saving.
* saving in CSV file is not implemented as such.
* not sure why we need $mode attribute and how it will be used

File: src/Persistence_CSV.php (hunk 1 above)
Comment (romaninsh) [inline src/Persistence_CSV.php]: added.

File: src/Persistence_CSV.php (hunk 2 above)
Comment (romaninsh) [inline src/Persistence_CSV.php]: fixed.

**Comment by romaninsh** [comment]
Dark, to answer your question about mode - the idea is to prevent user from writing into persistence if he has read from it.

Perhaps we could require user to specify if he wants to read or write to the file, but that can be improved later.

**Pull request merged**